src/west/manifest.py

```diff
@@ -67,6 +68,64 @@
 # Internal helpers
 #
 
+# Mapping of the architectures from platofrm.machine() to a common standard
+_ARCH_MAPPING = {
+    "x86_64": "amd64",
+    "aarch64": "arm64",
+}
+
+def _get_global_defitions() -> Dict[str, str]:
+    """
+    Get the default global 'os', 'arch', and 'platform' values.
+
+    Returns:
+        A dictionary containing the keys 'os', 'arch', and 'platform' and their
+        current values.
+    """
+    defs = {}
+    current_os = platform.system()
+    if current_os == "Windows":
+        defs["os"] = "windows"
+    elif current_os == "Darwin":
+        defs["os"] = "mac"
+    elif current_os == "Linux":
+        defs["os"] = "linux"
+    else:
+        raise RuntimeError("Unknown OS: " + current_os)
```

Comment:
Does this code run when you don't use the new feature?

Reply:
It did, thanks for catching it. I've moved it to a lazily initialized class so it only gets run if you're using the feature.